vxquery-core/src/test/java/org/apache/vxquery/datamodel/ObjectByteTest.java

```diff
@@ -0,0 +1,253 @@
+        }
+
+        //Test values
+        try {
+            op.getValue(tvpKey1, tvp);
+        } catch (SystemException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+        if (!FunctionHelper.arraysEqual(tvp, tvpValue1)) {
+            Assert.fail("Value is incorrect for the given key. Expected: A green door with valuetag: "
+                    + ValueTag.XS_STRING_TAG + " Got valuetag: " + tvp.getTag());
+        }
+        try {
+            op.getValue(tvpKey2, tvp);
+        } catch (SystemException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+        if (!FunctionHelper.arraysEqual(tvp, tvpValue2)) {
+            Assert.fail("Value is incorrect for the given key. Expected: 12.5 with valuetag: " + ValueTag.XS_DOUBLE_TAG
+                    + " Got valuetag: " + tvp.getTag());
+        }
+        try {
+            op.getValue(tvpKey3, tvp);
+        } catch (SystemException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+        if (!FunctionHelper.arraysEqual(tvp, tvpValue3)) {
+            Assert.fail("Value is incorrect for the given key. Expected: 100 with valuetag: " + ValueTag.XS_LONG_TAG
+                    + " Got valuetag: " + tvp.getTag());
+        }
+    }
+
+    @Test
+    public void testItemNotExistObject() {
+        // Build empty test Object
+        abvsResult.reset();
+        try {
+            ob.reset(abvsResult);
+            ob.finish();
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the Object pointable.");
+        }
+        tvp.set(abvsResult);
+        tvp.getValue(op);
+        try {
+            getTaggedValuePointable("key", tvpKey1);
+            op.getValue(tvpKey1, tvpValue1);
+        } catch (IOException | SystemException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+        XDMConstants.setFalse(tvp);
+        if (!FunctionHelper.arraysEqual(tvp, tvpValue1)) {
+            Assert.fail("Value is incorrect for the given key. Expected: false with valuetag: "
+                    + ValueTag.XS_BOOLEAN_TAG + " Got valuetag: " + tvp.getTag());
+        }
+
+        // Build test object
+        try {
+            // Add two items
+            ob.reset(abvsResult);
+            getTaggedValuePointable("name", tvpKey1);
+            getTaggedValuePointable("A green door", tvpValue1);
+            ob.addItem(tvpKey1, tvpValue1);
+            getTaggedValuePointable("price", tvpKey2);
+            getTaggedValuePointable(12.5, tvpValue2);
+            ob.addItem(tvpKey2, tvpValue2);
+            ob.finish();
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+
+        tvp.set(abvsResult);
+        tvp.getValue(op);
+        try {
+            getTaggedValuePointable("key", tvpKey3);
+            op.getValue(tvpKey3, tvpValue3);
```

Comment:
// Something like this...
if (op.getValue(tvpKey3, tvpValue3)) {
   // found key
}